### PR TITLE
append init segment data on audio track changes

### PR DESCRIPTION
**Pull request by mjneil**
required for [pull request](https://github.com/videojs/mux.js/pull/120)

### CODE REVIEW

File: src/html-media-source.js [hunk 1]
```diff
@@ -131,6 +131,8 @@ export default class HtmlMediaSource extends videojs.EventTarget {
         //      what stream is the video stream, rather than relying on videoTracks
         /* eslinst-enable */
 
+        sourceBuffer.appendInitSegment_ = true;
```
Comment (gesinger) [inline src/html-media-source.js]: Do we have to do this on all of 'change', 'addtrack', and 'removetrack'?

**Comment by gesinger** [comment]
Looks pretty good, but might be good to add some unit tests for when appendInitSegment_ is changed and that processPendingSegments_ adds the init segments to video and audio segment lists when appendInitSegment_ is true (and doesn't when false).

File: test/html.test.js [hunk 2]
```diff
@@ -438,6 +443,75 @@ function() {
   QUnit.equal(mp4Segments[0][1], 6, 'fragment contains the correct second byte');
 });
 
+QUnit.test(
+'only appends init segment for first segment or on audio changes',
```
Comment (gesinger) [inline test/html.test.js]: I know that audio changes change the appendInitSegment_ value, but I didn't see anything in the test that identifies audio changes.

Comment (mjneil) [inline test/html.test.js]: You're right. I've updated the test to trigger events on the player.

**Comment by gesinger** [comment]
LGTM

**Comment by imbcmdth** [comment]
LGTM

**Pull request merged**